architecture/src/androidLibMain/kotlin/lifecycle/getOrPutAndRemoveOnDestroyFromCache.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.architecture.lifecycle
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import kotlin.reflect.KClass
+
+/**
+ * @suppress
+ *
+ * Do not use directly, but use through [AppCompatActivity.getOrPutAndRemoveOnDestroyFromCache]
+ */
+val lifecycleAwareActivityCache = mutableMapOf<Pair<AppCompatActivity, KClass<*>>, Any>()
```

Comment:
Since the doc specifies that it should not be used directly and it is used only inside `getOrPutAndRemoveOnDestroyFromCache`, can it be an `internal`?

Reply:
No, due to `inline` :( 

Without inline we need an extra parameter for the type of class used. I prefered the cleaner API in the end.